pkg/blockbuilder/blockbuilder_test.go

```diff
@@ -718,19 +723,10 @@ func TestBlockBuilder_LongCatchupWithNoRecordsProcessed(t *testing.T) {
 		require.NoError(t, services.StopAndAwaitTerminated(ctx, bb))
 	})
 
-	// First cycle should happen normally.
-	select {
-	case <-compactCalled:
-	case <-time.After(5 * time.Second):
-		t.Fatal("first compaction did not happen within expected time")
-	}
-	// Second cycle is prone to getting stuck. So we wait for a while and then check if the
-	// second compaction happens.
-	select {
-	case <-compactCalled:
-	case <-time.After(15 * time.Second):
-		t.Fatal("second compaction did not happen within expected time")
-	}
+	time.Sleep(5 * time.Second)
+
+	// Nothing is consumed due to zero lag.
+	require.Zero(t, compactCalled.Load(), "expected skipping all records before lookback period")
```

Comment:
Do these test change correctly test the changes? I will have a proper look again.

Reply:
I think it does. Before the changes we tried to consume from the lag up until the lookback and called compact on an empty tsdb. Now, there is no "first" (empty) cycle, because we start consuming straight at the lookback record.